packages/jest-diff/src/diff_strings.js

```diff
@@ -133,6 +161,72 @@ const formatLine = (
   );
 };
 
+const NEWLINE_REGEXP = /\n/;
+
+// Given diff items from diff-match-patch,
+// return them split or merged corresponding to deleted and inserted lines.
+const alignDelInsDiffs = (diffs: DiffItems): DelInsDiffs => {
+  const diffsDel = [[]];
+  const diffsIns = [[]];
+
+  diffs.forEach(diff => {
+    const digit = diff[0];
+    const value = diff[1];
+
+    if (NEWLINE_REGEXP.test(value)) {
+      value.split('\n').forEach((line, i, lines) => {
+        // Omit any empty strings that split returns to separate newlines.
+        const subdiff = line.length !== 0 ? [digit, line] : null;
+
+        // If substring is unchanged, push onto deleted and inserted.
+        if (digit !== 1) {
+          if (subdiff !== null) {
+            diffsDel[diffsDel.length - 1].push(subdiff);
+          }
+          if (i !== lines.length - 1) {
+            diffsDel.push([]); // next line
+          }
+        }
+        if (digit !== -1) {
+          if (subdiff !== null) {
+            diffsIns[diffsIns.length - 1].push(subdiff);
+          }
+          if (i !== lines.length - 1) {
+            diffsIns.push([]); // next line
+          }
+        }
+      });
+    } else {
+      // If string is unchanged, push onto deleted and inserted.
+      if (digit !== 1) {
+        diffsDel[diffsDel.length - 1].push(diff);
+      }
+      if (digit !== -1) {
```

Comment:
`else if`

Reply:
Aha, `no-else-return` is not among the rules for this project :)